src/helm/benchmark/scenarios/medalign_scenario.py

```diff
@@ -80,61 +138,462 @@ def get_ehrs(path_to_ehrs: str) -> Dict[int, str]:
     return ehr_map
 
 
-def get_instructions(path_to_instructions: str) -> Dict[int, Dict[str, Union[int, str]]]:
+def get_tokenizer(tokenizer_name: str):
     """
-    Builds map from Instruction ID to instruction details
+    Returns a tokenizer based on the given tokenizer name.
 
-    The needed information for creating the map is accomplished by reading
-    a CSV file from the user-specified specified path.
+    Parameters:
+        tokenizer_name (str): The name of the tokenizer. Acceptable values are:
+            - "tiktoken"
+            - "chatgpt"
+            - "gpt-3.5-turbo"
+            - "gpt-4"
+            - "gpt-4-turbo"
+            - "gpt-4o"
+            - "cl100k_base"
+            - Any valid tokenizer name recognized by the transformers library.
 
-    The CSV file is expected to contain at least the following columns:
-    - instruction_id: The ID of the instruction.
-    - question: The text of the instruction.
-    - person_id: The ID of the associated patient.
-    - is_selected_ehr: A flag indicating whether the instruction is selected.
+    Returns:
+        Callable: The tokenizer instance.
+    """
+    if tokenizer_name.lower() in [
+        "tiktoken",
+        "chatgpt",
+        "gpt-3.5-turbo",
+        "gpt-4",
+        "gpt-4-turbo",
+        "gpt-4o",
+        "cl100k_base",
+    ]:
+        return tiktoken.get_encoding("cl100k_base")
+    print(tokenizer_name)
+    return transformers.AutoTokenizer.from_pretrained(tokenizer_name, legacy=False)
+
+
+def tag_rgx_expression(include, exclude):
+    """Create rgx expression to determine which tags should be included or excluded"""
+    if include:
+        return re.compile("|".join(include))
+    elif exclude:
+        return re.compile(f'^((?!({"|".join(exclude)})))')
+    return re.compile(".*")
+
+
+def fetch_nodes_with_tag(start_node, tag_str):
+    """Fetch nodes with certain tag value"""
+    return start_node.xpath(tag_str)
+
+
+def cast_dtype(i: str) -> Union[str, datetime.datetime, int, float]:
+    """Convert string to its appropriate type"""
+    try:
+        return ast.literal_eval(i)
```

Comment:
Optional, minor: Would prefer if you did a nested try-catch using `float()` and `int()` (in that order) rather than `literal_eval()`. But it looks like this is [mostly safe](https://docs.python.org/3/library/ast.html#ast.literal_eval).

Reply:
MedAlign is out of scope for review, so wontfix.